plugins/modules/vmware_dvs_host.py

```diff
@@ -271,7 +271,7 @@ def main():
         dict(
             esxi_hostname=dict(required=True, type='str'),
             switch_name=dict(required=True, type='str'),
-            vmnics=dict(required=True, type='list'),
+            vmnics=dict(required=False, type='list'),
```

Comment:
```suggestion
            vmnics=dict(required=False, type='list', default=[]),
```

I think you'll need to set the default to the empty list, otherwise I'm pretty sure line 182 will break because it tries to iterate over a `None`.